runners/spark/src/main/java/org/apache/beam/runners/spark/translation/SparkPCollectionView.java

```diff
@@ -0,0 +1,100 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.runners.spark.translation;
+
+import java.io.Serializable;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import org.apache.beam.runners.spark.coders.CoderHelpers;
+import org.apache.beam.runners.spark.util.BroadcastHelper;
+import org.apache.beam.sdk.coders.Coder;
+import org.apache.beam.sdk.util.WindowedValue;
+import org.apache.beam.sdk.values.PCollectionView;
+import org.apache.spark.api.java.JavaSparkContext;
+import scala.Tuple2;
+
+/**
+ * SparkPCollectionView is used to pass serialized views to lambdas.
+ */
+public class SparkPCollectionView implements Serializable {
+
+    // Holds the view --> broadcast mapping. Transient so it will be null from resume
+    private transient volatile Map<PCollectionView<?>, BroadcastHelper> broadcastHelperMap = null;
+
+    // Holds the Actual data of the views in serialize form
+    private Map<PCollectionView<?>,
+        Tuple2<byte[], Coder<Iterable<WindowedValue<?>>>>> pviews =
+            Collections.synchronizedMap(
+                new LinkedHashMap<PCollectionView<?>,
+                    Tuple2<byte[], Coder<Iterable<WindowedValue<?>>>>>());
+
+    // Driver only - during evaluation stage
+    void putPView(PCollectionView<?> view,
+                  Iterable<WindowedValue<?>> value,
+                  Coder<Iterable<WindowedValue<?>>> coder,
+                  JavaSparkContext context) {
+
+        pviews.put(view, new Tuple2(CoderHelpers.toByteArray(value, coder), coder));
+        // overwrite/create broadcast
+        getPCollectionView(view, context, true);
+    }
+
+    BroadcastHelper getPCollectionView(PCollectionView<?> view,
+                                       JavaSparkContext context) {
+        return getPCollectionView(view, context, false);
+    }
+
+    private BroadcastHelper getPCollectionView(PCollectionView<?> view,
+                                               JavaSparkContext context,
+                                               boolean overwrite) {
+        // initialize broadcastHelperMap if needed
+        if (broadcastHelperMap == null) {
+            synchronized (SparkPCollectionView.class) {
+                if (broadcastHelperMap == null) {
+                    broadcastHelperMap = new LinkedHashMap<>();
+                }
+            }
+        }
+
+        //lazily broadcast views
+        BroadcastHelper helper = broadcastHelperMap.get(view);
+        if (helper == null) {
+            synchronized (SparkPCollectionView.class) {
+                helper = broadcastHelperMap.get(view);
+                if (helper == null) {
+                    helper = getBroadcastHelper(view, context);
+                }
+            }
+        } else if (overwrite) {
+            synchronized (SparkPCollectionView.class) {
+                helper.unpersist(context);
```

Comment:
Can comment that `unpersist` can be blocking.. if we'd want to consider this in the future.